packages/core/src/libraries/component.ts

```diff
@@ -158,8 +155,9 @@ export function getComponentInfo(
     return;
   }
 
-  const screenshotPaths = (model.variations || [])
-    .map((v: VariationAsObject) => {
+  const smModel = Slices.toSM(model);
```

Comment:
Yeah it's because we have validation right before that fromn the FS, meaning that we have the object version.
I realise that sometimes we validate right away what comes from the FS and sometimes we cast.
If we want validation everywhere, it'll be the same problem as today. So either we throw and it won't alter the output type otherwise it can be invalid (meaning undefined or error) so we should handle this at every place where we read/write, something we don't do today because of the optimist cast

Reply:
@Miniplop Either we rework how we handle error with this PR or we keep the same behavior as today and we dedicate a PR to rework the error handling alone.